cmd/tink-worker/tink/tink.go

```diff
@@ -0,0 +1,70 @@
+package tink
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/pkg/errors"
+	pb "github.com/tinkerbell/tink/protos/workflow"
+)
+
+type Wdata struct {
+	Data WorkflowData
+}
+
+type WorkflowData struct {
+	WorkerID     string
+	WorkflowID   string
+	ActionStatus *pb.WorkflowActionStatus
+}
+
+// GetWorkflowContexts gets workflow context for worker id on success else returns error.
+func (w Wdata) GetWorkflowContexts(ctx context.Context, client pb.WorkflowServiceClient) (pb.WorkflowService_GetWorkflowContextsClient, error) {
+	if w.Data.WorkerID == "" {
+		return nil, errors.New("Empty string is not a valid worker id")
+	}
+
+	if client == nil || (reflect.ValueOf(client).IsNil()) {
+		return nil, errors.New("WorkflowServiceClient interface is not valid")
+	}
+
+	response, err := client.GetWorkflowContexts(ctx, &pb.WorkflowContextRequest{WorkerId: w.Data.WorkerID})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get workflow contexts")
```

Comment:
Maybe @jacobweinstock could weigh in on this - I think Go error messages should always start with a lower case letter. The contents of the error can contain capitalized letters as needed, but I think keeping as much of the error lower cased as possible helps with readability and searching during trouble shooting.

This blog post comes to mind, but it doesn't mention anything about capitalization: https://go.dev/blog/error-handling-and-go

Reply:
sure, I think that's something I've generally seen but I can't recall any one/place saying it's idiomatic. 


>When feasible, error strings should identify their origin, such as by having a prefix naming the operation or package that generated the error. For example, in package image, the string representation for a decoding error due to an unknown format is "image: unknown format".

--[ref](https://go.dev/doc/effective_go#:~:text=When%20feasible%2C%20error%20strings%20should%20identify%20their%20origin%2C%20such%20as%20by%20having%20a%20prefix%20naming%20the%20operation%20or%20package%20that%20generated%20the%20error.%20For%20example%2C%20in%20package%20image%2C%20the%20string%20representation%20for%20a%20decoding%20error%20due%20to%20an%20unknown%20format%20is%20%22image%3A%20unknown%20format%22.)